src/navigation/Navigator.tsx

```diff
@@ -1,242 +1,55 @@
-            defaultOptions(labels.general.header.overviewExercises, ArrowLeftCircleIconWhite, navigation, false)
-          }
           name='VocabularyList'
           component={VocabularyListScreen}
+          options={({ navigation }) => options(overviewExercises, navigation, false)}
         />
         <Stack.Screen
-          options={({ navigation }) =>
-            defaultOptions(labels.general.header.cancelExercise, CloseCircleIconWhite, navigation, false)
-          }
           name='WordChoiceExercise'
           component={WordChoiceExerciseScreen}
+          options={({ navigation }) => options(cancelExercise, navigation, true)}
         />
         <Stack.Screen
-          options={({ navigation }) =>
-            defaultOptions(labels.general.header.cancelExercise, CloseCircleIconWhite, navigation, false)
-          }
           name='ArticleChoiceExercise'
           component={ArticleChoiceExerciseScreen}
+          options={({ navigation }) => options(cancelExercise, navigation, true)}
         />
         <Stack.Screen
-          options={({ navigation }) =>
-            defaultOptions(labels.general.header.overviewExercises, CloseCircleIconWhite, navigation, false)
-          }
           name='WriteExercise'
           component={WriteExerciseScreen}
+          options={({ navigation }) => options(cancelExercise, navigation, true)}
         />
-        <Stack.Screen options={{ headerShown: false }} name='ExerciseFinished' component={ExerciseFinishedScreen} />
-        <Stack.Screen
-          options={{
-            headerLeft: () => null,
-            headerTitle: '',
-            headerRightContainerStyle: styles().headerRightContainer
-          }}
-          name='Result'
-          component={ResultScreen}
-        />
-        <Stack.Screen
-          options={({ navigation }) =>
-            defaultOptions(labels.results.resultsOverview, ArrowLeftCircleIconWhite, navigation, false)
-          }
-          name='ResultDetail'
-          component={ResultDetailScreen}
-        />
-        <Stack.Screen
-          options={({ navigation }) =>
-            defaultOptions(labels.general.header.overview, ArrowLeftCircleIconWhite, navigation, false)
-          }
-          name='AddCustomDiscipline'
-          component={AddCustomDisciplineScreen}
-        />
-        <Stack.Screen
-          options={({ navigation }) =>
-            defaultOptions(labels.general.header.overview, ArrowLeftCircleIconWhite, navigation, false)
-          }
-          name='Imprint'
-          component={ImprintScreen}
-        />
-        <Stack.Screen
-          options={({ navigation }) =>
-            defaultOptions(labels.general.header.overview, ArrowLeftCircleIconWhite, navigation, false)
-          }
-          name='ManageDisciplines'
-          component={ManageSelectionsScreen}
-        />
+        <Stack.Screen name='ExerciseFinished' component={ExerciseFinishedScreen} options={{ headerShown: false }} />
```

Comment:
I think we should also add Result and ResultDetails routes here, so the app has a valid state until the ticket, where they get removed is done.